flink-runtime/src/main/java/org/apache/flink/runtime/taskexecutor/JobLeaderService.java

```diff
@@ -251,90 +254,83 @@ private JobManagerLeaderListener(JobID jobId) {
 		}
 
 		public void stop() {
-			stopped = true;
+			synchronized (lock) {
+				stopped = true;
```

Comment:
break if `false`?

Reply:
Yes, but I think this wouldn't make much of a difference.